project-base/storefront/components/Forms/Radiobutton/Radiobutton.tsx

```diff
@@ -12,25 +12,22 @@ export type RadiobuttonProps = NativeProps & {
     value: any;
     checked: InputHTMLAttributes<HTMLInputElement>['checked'];
     label: ReactNode;
-    onChangeCallback?: (newValue: string | null) => void;
+    onClick?: (newValue: string | null) => void;
 };
 
 export const Radiobutton = forwardRef<HTMLInputElement, RadiobuttonProps>(
-    ({ label, onChangeCallback, onChange, id, name, checked, value, disabled, onBlur }, radiobuttonForwardedRef) => {
-        const onClickHandler: MouseEventHandler<HTMLInputElement> = useCallback(
-            (event) => {
-                if (!onChangeCallback) {
-                    return;
-                }
+    ({ label, onChange, id, name, checked, value, disabled, onBlur, onClick }, radiobuttonForwardedRef) => {
+        const onClickWithNull: MouseEventHandler<HTMLInputElement> = (event) => {
+            if (!onClick) {
+                return;
+            }
 
-                if (checked) {
-                    onChangeCallback(null);
-                } else {
-                    onChangeCallback(event.currentTarget.value);
-                }
-            },
-            [checked, onChangeCallback],
-        );
+            if (checked) {
+                onClick(null);
```

Comment:
I am sure there will be some motivation behind this change, but I don't see it. Can you explain please?

Reply:
Yeah, it was a process. I saw it being called `onChangeCallback`, which made no sense as it was used along with `onChange`, which would make the developer think it is the exact same thing. However, internally, `onClick` was used. Because of that, I thought it would be cleaner and more descriptive to show that it is actually `onClick`. Do you agree?

Reply:
That won't work. We need it to act like a checkbox, meaning that if I click on an already selected radiobutton, I want to uncheck it. Your code would not do anything in that case.

Reply:
agreed to keep